extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '../../../src/service';
+import {documentStateFor} from '../../../src/document-state';
+import {Observable} from '../../../src/observable';
+
+export function addListener(eventType, listner) {
+  return instrumentationServiceFor(window).addListener(eventType, listner);
+};
+
+
+/**
+ * Ignore Most of this class as it has not been thought through yet. It will
+ * change completely.
+ */
+class AnalyticsEvent {
+
+  constructor(type) {
+    this.type = type;
+  }
+}
+
+/** @private */
+class InstrumentationService {
+  /**
+   * @param {!Window} window
+   */
+  constructor(window) {
+    this.docState_ = documentStateFor(window);
+  }
+
+  addListener(eventType, listener) {
+
+    // TODO: Add support for clicks, timers, scroll etc.
+    if (eventType.toLowerCase() == 'visible') {
+      if (this.docState_.getVisibilityState() == 'visible') {
+        listener(new AnalyticsEvent('visible'));
+      } else {
+        this.docState_.onVisibilityChanged(() => {
```

Comment:
DocState is not a good fit here. We need to use `Viewer.isVisible` and `Viewer.onVisibilityChanged`.